[ShallowWater] Windows compilation

**Description**
Error during windows compilation

**Scope**
- Compilation
- ShallowWaterApplication

**Environment**
- OS: Windows 10

**Log**
```
Creating library E:/PROYECTOS/KRATOS/Kratos/build/Release/applications/ShallowWaterApplication/Release/KratosShallowWaterApplication.lib and object E:/PROYECTOS/KRATOS/Kratos/b
  uild/Release/applications/ShallowWaterApplication/Release/KratosShallowWaterApplication.exp
add_custom_strategies_to_python.obj : error LNK2019: unresolved external symbol "public: static void __cdecl Kratos::DerivativesRecoveryUtility<2>::CalculatePolynomialWeights(class 
Kratos::ModelPart &)" (?CalculatePolynomialWeights@?$DerivativesRecoveryUtility@$01@Kratos@@SAXAEAVModelPart@2@@Z) referenced in function "public: virtual void __cdecl Kratos::Resid
ualBasedAdamsMoultonScheme<class Kratos::UblasSpace<double,class boost::numeric::ublas::compressed_matrix<double,struct boost::numeric::ublas::basic_row_major<unsigned __int64,__int 
64>,0,class boost::numeric::ublas::unbounded_array<unsigned __int64,class std::allocator<unsigned __int64> >,class boost::numeric::ublas::unbounded_array<double,class std::allocator 
<double> > >,class boost::numeric::ublas::vector<double,class boost::numeric::ublas::unbounded_array<double,class std::allocator<double> > > >,class Kratos::UblasSpace<double,class  
boost::numeric::ublas::matrix<double,struct boost::numeric::ublas::basic_row_major<unsigned __int64,__int64>,class boost::numeric::ublas::unbounded_array<double,class std::allocator 
<double> > >,class boost::numeric::ublas::vector<double,class boost::numeric::ublas::unbounded_array<double,class std::allocator<double> > > > >::Initialize(class Kratos::ModelPart  
&)" (?Initialize@?$ResidualBasedAdamsMoultonScheme@V?$UblasSpace@NV?$compressed_matrix@NU?$basic_row_major@_K_J@ublas@numeric@boost@@$0A@V?$unbounded_array@_KV?$allocator@_K@std@@@2 
34@V?$unbounded_array@NV?$allocator@N@std@@@234@@ublas@numeric@boost@@V?$vector@NV?$unbounded_array@NV?$allocator@N@std@@@ublas@numeric@boost@@@234@@Kratos@@V?$UblasSpace@NV?$matrix 
@NU?$basic_row_major@_K_J@ublas@numeric@boost@@V?$unbounded_array@NV?$allocator@N@std@@@234@@ublas@numeric@boost@@V?$vector@NV?$unbounded_array@NV?$allocator@N@std@@@ublas@numeric@b 
oost@@@234@@2@@Kratos@@UEAAXAEAVModelPart@2@@Z) [E:\PROYECTOS\KRATOS\Kratos\build\Release\applications\ShallowWaterApplication\KratosShallowWaterApplication.vcxproj]
add_custom_utilities_to_python.obj : error LNK2001: unresolved external symbol "public: static void __cdecl Kratos::DerivativesRecoveryUtility<2>::CalculatePolynomialWeights(class K 
ratos::ModelPart &)" (?CalculatePolynomialWeights@?$DerivativesRecoveryUtility@$01@Kratos@@SAXAEAVModelPart@2@@Z) [E:\PROYECTOS\KRATOS\Kratos\build\Release\applications\ShallowWater 
Application\KratosShallowWaterApplication.vcxproj]
add_custom_strategies_to_python.obj : error LNK2019: unresolved external symbol "public: static void __cdecl Kratos::DerivativesRecoveryUtility<2>::RecoverLaplacian(class Kratos::Mo 
delPart &,class Kratos::Variable<class Kratos::array_1d<double,3> > const &,class Kratos::Variable<class Kratos::array_1d<double,3> > const &,unsigned __int64)" (?RecoverLaplacian@? 
$DerivativesRecoveryUtility@$01@Kratos@@SAXAEAVModelPart@2@AEBV?$Variable@V?$array_1d@N$02@Kratos@@@2@1_K@Z) referenced in function "public: virtual void __cdecl Kratos::ResidualBas 
edAdamsMoultonScheme<class Kratos::UblasSpace<double,class boost::numeric::ublas::compressed_matrix<double,struct boost::numeric::ublas::basic_row_major<unsigned __int64,__int64>,0, 
class boost::numeric::ublas::unbounded_array<unsigned __int64,class std::allocator<unsigned __int64> >,class boost::numeric::ublas::unbounded_array<double,class std::allocator<doubl 
e> > >,class boost::numeric::ublas::vector<double,class boost::numeric::ublas::unbounded_array<double,class std::allocator<double> > > >,class Kratos::UblasSpace<double,class boost: 
:numeric::ublas::matrix<double,struct boost::numeric::ublas::basic_row_major<unsigned __int64,__int64>,class boost::numeric::ublas::unbounded_array<double,class std::allocator<doubl 
e> > >,class boost::numeric::ublas::vector<double,class boost::numeric::ublas::unbounded_array<double,class std::allocator<double> > > > >::InitializeNonLinIteration(class Kratos::M 
odelPart &,class boost::numeric::ublas::compressed_matrix<double,struct boost::numeric::ublas::basic_row_major<unsigned __int64,__int64>,0,class boost::numeric::ublas::unbounded_arr 
ay<unsigned __int64,class std::allocator<unsigned __int64> >,class boost::numeric::ublas::unbounded_array<double,class std::allocator<double> > > &,class boost::numeric::ublas::vect 
or<double,class boost::numeric::ublas::unbounded_array<double,class std::allocator<double> > > &,class boost::numeric::ublas::vector<double,class boost::numeric::ublas::unbounded_ar 
ray<double,class std::allocator<double> > > &)" (?InitializeNonLinIteration@?$ResidualBasedAdamsMoultonScheme@V?$UblasSpace@NV?$compressed_matrix@NU?$basic_row_major@_K_J@ublas@nume 
ric@boost@@$0A@V?$unbounded_array@_KV?$allocator@_K@std@@@234@V?$unbounded_array@NV?$allocator@N@std@@@234@@ublas@numeric@boost@@V?$vector@NV?$unbounded_array@NV?$allocator@N@std@@@ 
ublas@numeric@boost@@@234@@Kratos@@V?$UblasSpace@NV?$matrix@NU?$basic_row_major@_K_J@ublas@numeric@boost@@V?$unbounded_array@NV?$allocator@N@std@@@234@@ublas@numeric@boost@@V?$vecto 
r@NV?$unbounded_array@NV?$allocator@N@std@@@ublas@numeric@boost@@@234@@2@@Kratos@@UEAAXAEAVModelPart@2@AEAV?$compressed_matrix@NU?$basic_row_major@_K_J@ublas@numeric@boost@@$0A@V?$u 
nbounded_array@_KV?$allocator@_K@std@@@234@V?$unbounded_array@NV?$allocator@N@std@@@234@@ublas@numeric@boost@@AEAV?$vector@NV?$unbounded_array@NV?$allocator@N@std@@@ublas@numeric@bo 
ost@@@567@2@Z) [E:\PROYECTOS\KRATOS\Kratos\build\Release\applications\ShallowWaterApplication\KratosShallowWaterApplication.vcxproj]
add_custom_utilities_to_python.obj : error LNK2001: unresolved external symbol "public: static void __cdecl Kratos::DerivativesRecoveryUtility<2>::RecoverLaplacian(class Kratos::Mod 
elPart &,class Kratos::Variable<class Kratos::array_1d<double,3> > const &,class Kratos::Variable<class Kratos::array_1d<double,3> > const &,unsigned __int64)" (?RecoverLaplacian@?$ 
DerivativesRecoveryUtility@$01@Kratos@@SAXAEAVModelPart@2@AEBV?$Variable@V?$array_1d@N$02@Kratos@@@2@1_K@Z) [E:\PROYECTOS\KRATOS\Kratos\build\Release\applications\ShallowWaterApplic 
ation\KratosShallowWaterApplication.vcxproj]
add_custom_utilities_to_python.obj : error LNK2019: unresolved external symbol "public: static void __cdecl Kratos::DerivativesRecoveryUtility<2>::Check(class Kratos::ModelPart &)"  
(?Check@?$DerivativesRecoveryUtility@$01@Kratos@@SAXAEAVModelPart@2@@Z) referenced in function "void __cdecl Kratos::Python::AddCustomUtilitiesToPython(class pybind11::module_ &)" ( 
?AddCustomUtilitiesToPython@Python@Kratos@@YAXAEAVmodule_@pybind11@@@Z) [E:\PROYECTOS\KRATOS\Kratos\build\Release\applications\ShallowWaterApplication\KratosShallowWaterApplication. 
vcxproj]
add_custom_utilities_to_python.obj : error LNK2019: unresolved external symbol "public: static void __cdecl Kratos::DerivativesRecoveryUtility<2>::RecoverDivergence(class Kratos::Mo 
delPart &,class Kratos::Variable<class Kratos::array_1d<double,3> > const &,class Kratos::Variable<double> const &,unsigned __int64)" (?RecoverDivergence@?$DerivativesRecoveryUtilit 
y@$01@Kratos@@SAXAEAVModelPart@2@AEBV?$Variable@V?$array_1d@N$02@Kratos@@@2@AEBV?$Variable@N@2@_K@Z) referenced in function "void __cdecl Kratos::Python::AddCustomUtilitiesToPython( 
class pybind11::module_ &)" (?AddCustomUtilitiesToPython@Python@Kratos@@YAXAEAVmodule_@pybind11@@@Z) [E:\PROYECTOS\KRATOS\Kratos\build\Release\applications\ShallowWaterApplication\K 
ratosShallowWaterApplication.vcxproj]
add_custom_utilities_to_python.obj : error LNK2019: unresolved external symbol "public: static void __cdecl Kratos::DerivativesRecoveryUtility<2>::RecoverGradient(class Kratos::Mode 
lPart &,class Kratos::Variable<double> const &,class Kratos::Variable<class Kratos::array_1d<double,3> > const &,unsigned __int64)" (?RecoverGradient@?$DerivativesRecoveryUtility@$0 
1@Kratos@@SAXAEAVModelPart@2@AEBV?$Variable@N@2@AEBV?$Variable@V?$array_1d@N$02@Kratos@@@2@_K@Z) referenced in function "void __cdecl Kratos::Python::AddCustomUtilitiesToPython(clas 
s pybind11::module_ &)" (?AddCustomUtilitiesToPython@Python@Kratos@@YAXAEAVmodule_@pybind11@@@Z) [E:\PROYECTOS\KRATOS\Kratos\build\Release\applications\ShallowWaterApplication\Krato 
sShallowWaterApplication.vcxproj]
add_custom_utilities_to_python.obj : error LNK2019: unresolved external symbol "public: static void __cdecl Kratos::DerivativesRecoveryUtility<2>::RecoverLaplacian(class Kratos::Mod 
elPart &,class Kratos::Variable<double> const &,class Kratos::Variable<double> const &,unsigned __int64)" (?RecoverLaplacian@?$DerivativesRecoveryUtility@$01@Kratos@@SAXAEAVModelPar 
t@2@AEBV?$Variable@N@2@1_K@Z) referenced in function "public: class pybind11::handle __cdecl <lambda_580c587a22527e5afbbb8ba8f76f378c>::operator()(struct pybind11::detail::function_ 
call &)const " (??R<lambda_580c587a22527e5afbbb8ba8f76f378c>@@QEBA?AVhandle@pybind11@@AEAUfunction_call@detail@2@@Z) [E:\PROYECTOS\KRATOS\Kratos\build\Release\applications\ShallowWa 
terApplication\KratosShallowWaterApplication.vcxproj]
add_custom_utilities_to_python.obj : error LNK2019: unresolved external symbol "public: static void __cdecl Kratos::DerivativesRecoveryUtility<3>::Check(class Kratos::ModelPart &)"  
(?Check@?$DerivativesRecoveryUtility@$02@Kratos@@SAXAEAVModelPart@2@@Z) referenced in function "void __cdecl Kratos::Python::AddCustomUtilitiesToPython(class pybind11::module_ &)" ( 
?AddCustomUtilitiesToPython@Python@Kratos@@YAXAEAVmodule_@pybind11@@@Z) [E:\PROYECTOS\KRATOS\Kratos\build\Release\applications\ShallowWaterApplication\KratosShallowWaterApplication. 
vcxproj]
add_custom_utilities_to_python.obj : error LNK2019: unresolved external symbol "public: static void __cdecl Kratos::DerivativesRecoveryUtility<3>::CalculatePolynomialWeights(class K
ratos::ModelPart &)" (?CalculatePolynomialWeights@?$DerivativesRecoveryUtility@$02@Kratos@@SAXAEAVModelPart@2@@Z) referenced in function "void __cdecl Kratos::Python::AddCustomUtili 
tiesToPython(class pybind11::module_ &)" (?AddCustomUtilitiesToPython@Python@Kratos@@YAXAEAVmodule_@pybind11@@@Z) [E:\PROYECTOS\KRATOS\Kratos\build\Release\applications\ShallowWater 
Application\KratosShallowWaterApplication.vcxproj]
add_custom_utilities_to_python.obj : error LNK2019: unresolved external symbol "public: static void __cdecl Kratos::DerivativesRecoveryUtility<3>::RecoverDivergence(class Kratos::Mo 
delPart &,class Kratos::Variable<class Kratos::array_1d<double,3> > const &,class Kratos::Variable<double> const &,unsigned __int64)" (?RecoverDivergence@?$DerivativesRecoveryUtilit 
y@$02@Kratos@@SAXAEAVModelPart@2@AEBV?$Variable@V?$array_1d@N$02@Kratos@@@2@AEBV?$Variable@N@2@_K@Z) referenced in function "void __cdecl Kratos::Python::AddCustomUtilitiesToPython( 
class pybind11::module_ &)" (?AddCustomUtilitiesToPython@Python@Kratos@@YAXAEAVmodule_@pybind11@@@Z) [E:\PROYECTOS\KRATOS\Kratos\build\Release\applications\ShallowWaterApplication\K 
ratosShallowWaterApplication.vcxproj]
add_custom_utilities_to_python.obj : error LNK2019: unresolved external symbol "public: static void __cdecl Kratos::DerivativesRecoveryUtility<3>::RecoverGradient(class Kratos::Mode 
lPart &,class Kratos::Variable<double> const &,class Kratos::Variable<class Kratos::array_1d<double,3> > const &,unsigned __int64)" (?RecoverGradient@?$DerivativesRecoveryUtility@$0 
2@Kratos@@SAXAEAVModelPart@2@AEBV?$Variable@N@2@AEBV?$Variable@V?$array_1d@N$02@Kratos@@@2@_K@Z) referenced in function "void __cdecl Kratos::Python::AddCustomUtilitiesToPython(clas 
s pybind11::module_ &)" (?AddCustomUtilitiesToPython@Python@Kratos@@YAXAEAVmodule_@pybind11@@@Z) [E:\PROYECTOS\KRATOS\Kratos\build\Release\applications\ShallowWaterApplication\Krato 
sShallowWaterApplication.vcxproj]
add_custom_utilities_to_python.obj : error LNK2019: unresolved external symbol "public: static void __cdecl Kratos::DerivativesRecoveryUtility<3>::RecoverLaplacian(class Kratos::Mod 
elPart &,class Kratos::Variable<double> const &,class Kratos::Variable<double> const &,unsigned __int64)" (?RecoverLaplacian@?$DerivativesRecoveryUtility@$02@Kratos@@SAXAEAVModelPar 
t@2@AEBV?$Variable@N@2@1_K@Z) referenced in function "public: __cdecl <lambda_490da4b9624f1c03cbaa1a69d65c7dc2>::operator()(class Kratos::ModelPart &,class Kratos::Variable<double>  
const &,class Kratos::Variable<double> const &,unsigned __int64)const " (??R<lambda_490da4b9624f1c03cbaa1a69d65c7dc2>@@QEBA@AEAVModelPart@Kratos@@AEBV?$Variable@N@2@1_K@Z) [E:\PROYE 
CTOS\KRATOS\Kratos\build\Release\applications\ShallowWaterApplication\KratosShallowWaterApplication.vcxproj]
add_custom_utilities_to_python.obj : error LNK2019: unresolved external symbol "public: static void __cdecl Kratos::DerivativesRecoveryUtility<3>::RecoverLaplacian(class Kratos::Mod 
elPart &,class Kratos::Variable<class Kratos::array_1d<double,3> > const &,class Kratos::Variable<class Kratos::array_1d<double,3> > const &,unsigned __int64)" (?RecoverLaplacian@?$ 
DerivativesRecoveryUtility@$02@Kratos@@SAXAEAVModelPart@2@AEBV?$Variable@V?$array_1d@N$02@Kratos@@@2@1_K@Z) referenced in function "public: __cdecl <lambda_1f96fbacaddb2965df798ef06 
09e893f>::operator()(class Kratos::ModelPart &,class Kratos::Variable<class Kratos::array_1d<double,3> > const &,class Kratos::Variable<class Kratos::array_1d<double,3> > const &,un 
signed __int64)const " (??R<lambda_1f96fbacaddb2965df798ef0609e893f>@@QEBA@AEAVModelPart@Kratos@@AEBV?$Variable@V?$array_1d@N$02@Kratos@@@2@1_K@Z) [E:\PROYECTOS\KRATOS\Kratos\build\ 
Release\applications\ShallowWaterApplication\KratosShallowWaterApplication.vcxproj]
E:\PROYECTOS\KRATOS\Kratos\build\Release\applications\ShallowWaterApplication\Release\KratosShallowWaterApplication.pyd : fatal error LNK1120: 12 unresolved externals [E:\PROYECTOS\ 
KRATOS\Kratos\build\Release\applications\ShallowWaterApplication\KratosShallowWaterApplication.vcxproj]
```

--- a/applications/ShallowWaterApplication/custom_utilities/derivatives_recovery_utility.h
+++ b/applications/ShallowWaterApplication/custom_utilities/derivatives_recovery_utility.h
@@ -48,7 +48,7 @@ class ModelPart;
  *    trans(a) = a_i
  */
 template<std::size_t TDim>
-class DerivativesRecoveryUtility
+class KRATOS_API(SHALLOW_WATER_APPLICATION) DerivativesRecoveryUtility
 {
 public:
     ///@name Type Definitions

